app/service/data_svc.py

```diff
@@ -323,23 +323,42 @@ async def _load_payloads(self, plugin):
-            payload_config[PAYLOADS_CONFIG_SPECIAL_KEY].update(special_payloads)
-            payload_config[PAYLOADS_CONFIG_EXTENSIONS_KEY].update(extensions)
-        self._update_payload_config(payload_config)
-
-    def _update_payload_config(self, updates):
+            standard_payloads = data[0].get(PAYLOADS_CONFIG_STANDARD_KEY, dict())
+            payload_config[PAYLOADS_CONFIG_STANDARD_KEY].update(standard_payloads)
+            if special_payloads:
+                await self._apply_special_payload_hooks(special_payloads)
+                payload_config[PAYLOADS_CONFIG_SPECIAL_KEY].update(special_payloads)
+            if extensions:
+                await self._apply_special_extension_hooks(extensions)
+                payload_config[PAYLOADS_CONFIG_EXTENSIONS_KEY].update(extensions)
+        self._update_payload_config(payload_config, plugin.name)
+
+    def _update_payload_config(self, updates, curr_plugin_name):
         payload_config = self.get_config(name='payloads')
-        payload_config[PAYLOADS_CONFIG_STANDARD_KEY] = {**payload_config.get(PAYLOADS_CONFIG_STANDARD_KEY, dict()),
-                                                        **updates.get(PAYLOADS_CONFIG_STANDARD_KEY, dict())}
-        payload_config[PAYLOADS_CONFIG_SPECIAL_KEY] = {**payload_config.get(PAYLOADS_CONFIG_SPECIAL_KEY, dict()),
-                                                       **updates.get(PAYLOADS_CONFIG_SPECIAL_KEY, dict())}
-        payload_config[PAYLOADS_CONFIG_EXTENSIONS_KEY] = {**payload_config.get(PAYLOADS_CONFIG_EXTENSIONS_KEY, dict()),
-                                                          **updates.get(PAYLOADS_CONFIG_EXTENSIONS_KEY, dict())}
+        curr_standard_payloads = payload_config.get(PAYLOADS_CONFIG_STANDARD_KEY, dict())
+        curr_special_payloads = payload_config.get(PAYLOADS_CONFIG_SPECIAL_KEY, dict())
+        curr_extensions = payload_config.get(PAYLOADS_CONFIG_EXTENSIONS_KEY, dict())
+        new_standard_payloads = updates.get(PAYLOADS_CONFIG_STANDARD_KEY, dict())
+        new_special_payloads = updates.get(PAYLOADS_CONFIG_SPECIAL_KEY, dict())
+        new_extensions = updates.get(PAYLOADS_CONFIG_EXTENSIONS_KEY, dict())
+
+        self._check_payload_overlaps(curr_standard_payloads, new_standard_payloads, PAYLOADS_CONFIG_STANDARD_KEY,
+                                     curr_plugin_name)
+        self._check_payload_overlaps(curr_special_payloads, new_special_payloads, PAYLOADS_CONFIG_SPECIAL_KEY,
+                                     curr_plugin_name)
+        self._check_payload_overlaps(curr_extensions, new_extensions, PAYLOADS_CONFIG_EXTENSIONS_KEY, curr_plugin_name)
+
+        payload_config[PAYLOADS_CONFIG_STANDARD_KEY] = {**curr_standard_payloads, **new_standard_payloads}
+        payload_config[PAYLOADS_CONFIG_SPECIAL_KEY] = {**curr_special_payloads, **new_special_payloads}
+        payload_config[PAYLOADS_CONFIG_EXTENSIONS_KEY] = {**curr_extensions, **new_extensions}
```

Comment:
Could use `.update()` instead, because of how the code is structured now.

Reply:
`.update()` modifies in place, whereas `{**,**}` will give us a new dict, which allows me to handle cases where maybe `payload_config` doesn't yet have a particular section. Like if `payload_config[PAYLOADS_CONFIG_EXTENSIONS_KEY]` returns `None`, I don't want to call `.update()` on that, and it'll get too verbose to add checks there.